Generate IPC serialization for WebKit::PrintInfo

zdobersek opened the pull request:
#### c2381fe366932db961b453199ca80d162f0e40a0
<pre>
Generate IPC serialization for WebKit::PrintInfo
<a href="https://bugs.webkit.org/show_bug.cgi?id=264274">https://bugs.webkit.org/show_bug.cgi?id=264274</a>

Reviewed by Michael Catanzaro.

Provide serialization input file for the WebKit::PrintInfo struct, removing the
need for custom encode/decode methods.

For the GTK port, the PrintMode enumeration is turned into the scoped equivalent
along with the necessary changes at points of use. ArgumentCoder specializations
are also provided for the two types that are GRefPtr-wrapped.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/Shared/PrintInfo.cpp:
(WebKit::PrintInfo::PrintInfo):
(WebKit::PrintInfo::encode const): Deleted.
(WebKit::PrintInfo::decode): Deleted.
* Source/WebKit/Shared/PrintInfo.h:
* Source/WebKit/Shared/PrintInfo.serialization.in: Added.
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPrintSettings&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPrintSettings&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPageSetup&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPageSetup&gt;&gt;::decode):
(IPC::encode): Deleted.
(IPC::decode): Deleted.
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewPrintFrame):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkitPrintOperationPrintPagesForFrame):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:
(WebKit::WebPrintOperationGtk::PrintPagesData::PrintPagesData):
(WebKit::WebPrintOperationGtk::print):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: <a href="https://commits.webkit.org/270377@main">https://commits.webkit.org/270377@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d12050149b11bb76f753624e5db92022f6f0faf5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25172 "9 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3713 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26428 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27287 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23096 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5425 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1149 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23340 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25415 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2741 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21723 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27866 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2426 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28786 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22973 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23011 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26612 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2376 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/668 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3723 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6068 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2813 "Built successfully") | | [⏳ 🛠 jsc-mips ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2708 "Built successfully") | | [⏳ 🧪 jsc-mips-tests ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/Shared/PrintInfo.serialization.in:
```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2020 Apple Inc. All rights reserved.
```
Looks like a copy/paste error?

zdobersek replied:
The serialization specification is identical to the code in the header, that's why I just copied the copyright to match the corresponding header.

zdobersek committed:
Generate IPC serialization for WebKit::PrintInfo
https://bugs.webkit.org/show_bug.cgi?id=264274

Reviewed by Michael Catanzaro.

Provide serialization input file for the WebKit::PrintInfo struct, removing the
need for custom encode/decode methods.

For the GTK port, the PrintMode enumeration is turned into the scoped equivalent
along with the necessary changes at points of use. ArgumentCoder specializations
are also provided for the two types that are GRefPtr-wrapped.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/Shared/PrintInfo.cpp:
(WebKit::PrintInfo::PrintInfo):
(WebKit::PrintInfo::encode const): Deleted.
(WebKit::PrintInfo::decode): Deleted.
* Source/WebKit/Shared/PrintInfo.h:
* Source/WebKit/Shared/PrintInfo.serialization.in: Added.
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
(IPC::ArgumentCoder<GRefPtr<GtkPrintSettings>>::encode):
(IPC::ArgumentCoder<GRefPtr<GtkPrintSettings>>::decode):
(IPC::ArgumentCoder<GRefPtr<GtkPageSetup>>::encode):
(IPC::ArgumentCoder<GRefPtr<GtkPageSetup>>::decode):
(IPC::encode): Deleted.
(IPC::decode): Deleted.
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewPrintFrame):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkitPrintOperationPrintPagesForFrame):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:
(WebKit::WebPrintOperationGtk::PrintPagesData::PrintPagesData):
(WebKit::WebPrintOperationGtk::print):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: https://commits.webkit.org/270377@main
